pkg/services/app/add_test.go

```diff
@@ -100,7 +115,44 @@ var _ = Describe("Add", func() {
 		Expect(deployKey).To(Equal([]byte("deploy key")))
 	})
 
+	Describe("checks for existing deploy key before creating secret", func() {
+		It("looks up deploy key and skips creating secret if found", func() {
+			defaultParams.SourceType = string(app.SourceTypeGit)
+
+			gitProviders.DeployKeyExistsStub = func(s1, s2 string) (bool, error) {
+				deployKeyLookups += 1
+				return true, nil
+			}
+
+			err := appSrv.Add(defaultParams)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(deployKeyUploads).To(Equal(0))
```

Comment:
To keep consistent with the rest of the tests you can use the `CallCount` methods, so you can get rid of `deployKeyLookups` and `deployKeyLookups` and only implement the custom Stub when it returns something different from the default values.

In this case, it could be:

```go
err := appSrv.Add(defaultParams)
Expect(err).ShouldNot(HaveOccurred())

Expect(gitProviders.DeployKeyExistsCallCount()).To(Equal(1))

Expect(fluxClient.CreateSecretGitCallCount()).To(Equal(0))
Expect(gitProviders.UploadDeployKeyCallCount()).To(Equal(0))
```

Reply:
I looked at that. That isn't what "CallCount" does. I wish it were. Here's the definition from the gitops fake:

```
func (fake *FakeGitProviderHandler) UploadDeployKeyCallCount() int {
	fake.uploadDeployKeyMutex.RLock()
	defer fake.uploadDeployKeyMutex.RUnlock()
	return len(fake.uploadDeployKeyArgsForCall)
}
```

It actually is just counting the args passed to the call. Justin and I ran into this last week.

Reply:
Sorry, missed the fluxClient reference here.